plugins/DevicesDetection/DevicesDetection.php

```diff
@@ -49,4 +50,33 @@ public function extendVisitorDetails(&$visitor, $details)
         $visitor['browserVersion']           = $instance->getBrowserVersion();
     }
 
+    public function renameUserSettingsModuleAndAction(&$module, &$action)
+    {
+        $movedMethods = array(
+            'index' => 'software',
+            'getBrowser' => 'getBrowsers',
+            'getBrowserVersion' => 'getBrowserVersions',
+            'getMobileVsDesktop' => 'getType',
+            'getOS' => 'getOsVersions',
+            'getOSFamily' => 'getOsFamilies',
+            'getBrowserType' => 'getBrowserEngines',
+        );
+
+        if ($module == 'UserSettings' && array_key_exists($action, $movedMethods)) {
+            $module = 'DevicesDetection';
+            $action = $movedMethods[$action];
+        }
+
+        if ($module == 'UserSettings' && ($action == 'getResolution' || $action == 'getConfiguration')) {
+            $module = 'Resolution';
```

Comment:
Should this one be in the `Resolution` plugin and the ones below in the other plugin? The plugins might not be enabled and moving them to the actual plugins will avoid having to check whether they are enabled here. 

Of course if eg `DevicePlugins` is disabled the BC for `getPlugin` doesn't work but in this case we should rather respect the users choice to have the plugin disabled